### PR TITLE
Trwalke/broker prompt

### PR DESCRIPTION
**Pull request by trwalke**
Fix for https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/issues/2133

Sent a preview package to customer and this fix has solved the issue for them.

### CODE REVIEW

File: src/client/Microsoft.Identity.Client/Platforms/Android/Broker/BrokerRequest.cs [hunk 1]
```diff
@@ -65,7 +65,17 @@ public Uri RedirectUri
             AcquireTokenInteractiveParameters acquireTokenInteractiveParameters)
         {
             BrokerRequest br = FromAuthenticationParameters(authenticationRequestParameters);
-            br.Prompt = acquireTokenInteractiveParameters.Prompt.PromptValue.ToUpperInvariant();
+            var prompt = acquireTokenInteractiveParameters.Prompt.PromptValue;
```
Comment (bgavrilMS) [inline src/client/Microsoft.Identity.Client/Platforms/Android/Broker/BrokerRequest.cs:68]: nit: you can compare Prompt objects directly as they have `==` overridden. Pretty obscure, not sure who decided that :(

File: src/client/Microsoft.Identity.Client/Platforms/Android/Broker/BrokerRequest.cs [hunk 2]
```diff
@@ -65,7 +65,17 @@ public Uri RedirectUri
...
+
+            if (prompt == Client.Prompt.NoPrompt.PromptValue || prompt == Client.Prompt.NotSpecified.PromptValue)
```
Comment (bgavrilMS) [inline src/client/Microsoft.Identity.Client/Platforms/Android/Broker/BrokerRequest.cs:70]: Can you apply this fix to iOS as well? If unsure about the value needed, then use the old "select_account"

**Review by bgavrilMS — CHANGES_REQUESTED** [review]
should apply to iOS broker as well

File: src/client/Microsoft.Identity.Client/Platforms/Android/Broker/BrokerRequest.cs (hunk 2 above)
Comment (henrik-me) [reply src/client/Microsoft.Identity.Client/Platforms/Android/Broker/BrokerRequest.cs]: why select account?

---
In reply to: [515239838](https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/pull/2154#discussion_r515239838) [](ancestors = 515239838)

File: src/client/Microsoft.Identity.Client/Platforms/Android/Broker/BrokerRequest.cs [hunk 3]
```diff
@@ -65,7 +65,17 @@ public Uri RedirectUri
             AcquireTokenInteractiveParameters acquireTokenInteractiveParameters)
         {
             BrokerRequest br = FromAuthenticationParameters(authenticationRequestParameters);
-            br.Prompt = acquireTokenInteractiveParameters.Prompt.PromptValue.ToUpperInvariant();
+            var prompt = acquireTokenInteractiveParameters.Prompt;
+
+            if (prompt == Client.Prompt.NoPrompt || prompt == Client.Prompt.NotSpecified)
+            {
+                br.Prompt = "NONE";
```
Comment (bgavrilMS) [inline src/client/Microsoft.Identity.Client/Platforms/Android/Broker/BrokerRequest.cs:72]: Isn't this a breaking change?

Comment (trwalke) [reply src/client/Microsoft.Identity.Client/Platforms/Android/Broker/BrokerRequest.cs:72]: we used to default to "none" for broker a long time ago but I think the behavior for "none" and "select_account" is ultimately the same. Ill switch to select account.

**Review by bgavrilMS — APPROVED** [review]

**Pull request merged**